contracts/src/v0.8/automation/tests/LogUpkeepCounter.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.6;
+
+contract LogUpkeepCounter {
+  event PerformingUpkeep(
+    address indexed from,
+    uint256 initialBlock,
+    uint256 lastBlock,
+    uint256 previousBlock,
+    uint256 counter
+  );
+
+  /**
+   * @dev we include multiple event types for testing various filters, signatures, etc
+   */
+  event Trigger(); // 0x3d53a39550e04688065827f3bb86584cb007ab9ebca7ebd528e7301c9c31eb5d
+  event Trigger(uint256 a); // 0x57b1de35764b0939dde00771c7069cdf8d6a65d6a175623f19aa18784fd4c6da
+  event Trigger(uint256 a, uint256 b); // 0x1da9f70fe932e73fba9374396c5c0b02dbd170f951874b7b4afabe4dd029a9c8
+  event Trigger(uint256 a, uint256 b, uint256 c); // 0x5121119bad45ca7e58e0bdadf39045f5111e93ba4304a0f6457a3e7bc9791e71
+
+  uint256 public testRange;
+  uint256 public lastBlock;
+  uint256 public previousPerformBlock;
+  uint256 public initialBlock;
+  uint256 public counter;
+
+  constructor(uint256 _testRange) {
+    testRange = _testRange;
+    previousPerformBlock = 0;
+    lastBlock = block.number;
+    initialBlock = 0;
+    counter = 0;
+  }
+
+  function start() public {
```

Comment:
Good coverage of existing scenarios, but we might need additional function that accepts the parameters to emit Trigger, then we could perform any case from outside the contract

Reply:
I will save that for a later PR when we have more concrete demands. Hopefully this is good enough for now :)